core/src/com/unciv/logic/automation/NextTurnAutomation.kt

```diff
@@ -687,17 +690,33 @@ object NextTurnAutomation {
     }
 
     private fun motivationToAttack(civInfo: CivilizationInfo, otherCiv: CivilizationInfo): Int {
+        if(civInfo.cities.isEmpty() || otherCiv.cities.isEmpty()) return 0
         val baseForce = 30f
+        
         val ourCombatStrength = civInfo.getStatForRanking(RankingType.Force).toFloat() + baseForce
         var theirCombatStrength = otherCiv.getStatForRanking(RankingType.Force).toFloat() + baseForce
 
         //for city-states, also consider there protectors
-        if(otherCiv.isCityState() and otherCiv.getProtectorCivs().isNotEmpty()) {
+        if (otherCiv.isCityState() and otherCiv.getProtectorCivs().isNotEmpty()) {
             theirCombatStrength += otherCiv.getProtectorCivs().sumOf{it.getStatForRanking(RankingType.Force)}
         }
 
         if (theirCombatStrength > ourCombatStrength) return 0
 
+        val closestCities = getClosestCities(civInfo, otherCiv)
+        val ourCity = closestCities.city1
+        val theirCity = closestCities.city2
+        
+        if (civInfo.getCivUnits().filter { it.isMilitary() }.none {
+                val damageRecievedWhenAttacking = 
+                    BattleDamage.calculateDamageToAttacker(
+                        MapUnitCombatant(it),
+                        CityCombatant(theirCity)
+                    )
+                damageRecievedWhenAttacking < 100
+            })
+                return 0 // You don't have any units that can attack this city without dying, don't declare war.
```

Comment:
Ranged units with range 3 could... But then if they fail this test their damage inflicted would likely be negligible too.

Reply:
Their damage to attacker would be 0, they would pass
To fail this you would have to have no ranged units at all, which does happen